src/components/requester.js

```diff
@@ -28,16 +30,38 @@ module.exports = class Requester extends Monitorable(Configurable(Component)) {
     }
 
     send(...args) {
-        if (args.length == 1 || typeof args[args.length - 1] != 'function') {
-            return new Promise((resolve, reject) => {
-                this.sock.send(...args, (err, res) => {
-                    if (err) return reject(err);
-                    resolve(res);
-                });
+        const cb = args[args.length - 1];
+        const hasCallback = 'function' == typeof cb;
+
+        const sockSend = (args) => Promise.race([
+            this.sockSend(args),
+            this.createTimeout(args),
+        ]);
+
+        if (hasCallback)
+            return sockSend(args.slice(0, -1))
+                .then((res) => cb(null, res))
+                .catch(cb);
+
+        return sockSend(args);
+    }
+
+    sockSend(args) {
+        return new Promise((resolve, reject) => {
+            this.sock.send(...args, (err, res) => {
+                if (err) return reject(err);
+                resolve(res);
             });
-        }
+        });
+    }
+
+    createTimeout([data = {}]) {
```

Comment:
createTimeout and sockSend shouldn't appear on the public class interface. can you make them local and therefore private?

Reply:
I moved this into module scope.